packages/formatters/src/github-actions.ts

```diff
@@ -0,0 +1,42 @@
+import { relative } from '@stoplight/path';
+import { DiagnosticSeverity, Dictionary } from '@stoplight/types';
+import { Formatter } from './types';
+
+const OUTPUT_TYPES: Dictionary<string, DiagnosticSeverity> = {
+  [DiagnosticSeverity.Error]: 'error',
+  [DiagnosticSeverity.Warning]: 'warning',
+  [DiagnosticSeverity.Information]: 'notice',
+  [DiagnosticSeverity.Hint]: 'notice',
+};
+
+type OutputParams = {
+  title?: string;
+  file: string;
+  col?: number;
+  endColumn?: number;
+  line?: number;
+  endLine?: number;
+};
+
+export const githubActions: Formatter = results => {
+  return results
+    .map(result => {
+      // GitHub Actions requires relative path for annotations, determining from working directory here
+      const file = relative(process.cwd(), result.source ?? '');
+      const params: OutputParams = {
+        title: result.code.toString(),
+        file,
+        col: result.range.start.character,
+        endColumn: result.range.end.character,
+        line: result.range.start.line,
+        endLine: result.range.end.line,
+      };
+
+      const paramsString = Object.entries(params)
+        .map(p => p.join('='))
+        .join(',');
+
+      return `::${OUTPUT_TYPES[result.severity]} ${paramsString}::${result.message}`;
```

Comment:
Do new lines in `result.message` need to be escaped somehow? At times, `result.message` may contain new lines if people define it that way in their ruleset

Reply:
@P0lip Yes, we need to escape newlines. As described in https://github.com/actions/toolkit/issues/193, we can use `%0A` URL-encoded character here.
Fixed in [77cddc0](https://github.com/stoplightio/spectral/pull/2508/commits/77cddc005d1b47f58e88cd9371c01f78665a1283).